src/validators/habit.validator.js

```
@@ -12,201 +13,146 @@ const isValidTimeZone = (value) => {
-    ) {
-      throw new Error(
-        'Request body must contain at least one valid field to update (name, icon, frequency, startDate, endDate).'
-      );
-    }
-    return true;
+// Validation schemas
+const createHabitSchema = z.object({
+  name: z.string().trim().min(1, 'Habit name is required.'),
+  icon: z.string().trim().optional(),
+  frequency: frequencySchema,
+  startDate: z.string().refine((date) => !isNaN(new Date(date).getTime()), {
```

ESLint

[error] Unexpected use of 'isNaN'. Use Number.isNaN instead https://github.com/airbnb/javascript#standard-library--isnan

(no-restricted-globals)